src/shared/generator.c

```diff
@@ -384,7 +384,7 @@ int generator_write_timeouts(
 
         r = fstab_filter_options(opts, "comment=systemd.device-timeout\0"
                                        "x-systemd.device-timeout\0",
-                                 NULL, &timeout, NULL, filtered);
+                                 NULL, &timeout, NULL, ret_filtered);
```

Comment:
Not your fault, but `ret_filtered` is set even if this function failed below.
If you touch this function, then please also fix that.

Reply:
(So that this can be resolved too)